src/main/java/bowling/domain/frame/FinalFrame.java

```diff
@@ -88,9 +84,44 @@ public StateHistory getStateHistory() {
         return stateHistory;
     }
 
-    private void assertFelledPin(int felledPin) {
-        if(felledPin > MAX_FELLED_PIN_COUNT || felledPin < MIN_FELLED_PIN_COUNT) {
-            throw new IllegalArgumentException(WRONG_FELLED_PIN);
+    @Override
+    public int getScore() {
+        if (!isEndedFrame()) {
+            return CAN_NOT_CALCULATE_SCORE;
         }
+
+        Score score = ((Finished) state).createScore();
```

Comment:
아래의 코드로 개선해 보면 어떨까요?

```suggestion
        Score score = state.createScore();
```